devito/types/basic.py

```diff
@@ -701,6 +701,7 @@ def grid(self):
 
     def _infer_dims(self):
         grids = {getattr(c, 'grid', None) for c in self.flat()} - {None}
+        grids = {g.root for g in grids}
```

Comment:
might be no girds here too

Reply:
Again, this is fine, it will just result in an empty set.